cmake/Modules/FindCustomScalapack.cmake

```diff
@@ -0,0 +1,124 @@
+# Distributed under the OSI-approved BSD 2-Clause License.
+#
+# Copyright (C) 2020 DFTB+ developers group
+#
+
+#[=======================================================================[.rst:
+FindCustomScalapack
+-------------------
+
+Finds the ScaLAPACK library.
+
+This is a simple auto-detection module for the ScaLAPACK library (looks
+basically for a library with the name 'scalapack'). It also assumes that
+ScaLAPACK is MPI-based and defines a respective dependency on
+``MPI::MPI_Fortran.``
+
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+This module provides the following imported target, if found:
+
+``Scalapack::Scalapack``
+  The ScaLAPACK library
+
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This module will define the following variable:
+
+``SCALAPACK_FOUND``
+  True if the system has the SCALAPACK library
+
+
+Cache variables
+^^^^^^^^^^^^^^^
+
+The following cache variables may be set to influence the library detection:
+
+``SCALAPACK_DETECTION``
+  Whether ScaLAPACK libraries should be detected (default: True). If set to False,
+  the settings in ``SCALAPACK_LIBRARY`` will be used without any further checks.
+
+``SCALAPACK_LIBRARY``
+  Customized ScaLAPACK library/libraries to use.  If no SCALAPACK library is
+  required (e.g. the linker automatically links it) set
+  ``SCALAPACK_LIBRARY="NONE"``. If not set or empty, it will use
+  ``find_package(scalapack)`` to find the scalapack library (relying on the
+  existence of a CMake export file for scalapack). Otherwise, the listed
+  libraries will be checked for existence (unless disabled in
+  ``SCALAPACK_DETECTION``) and the variable is overwritten to contain the
+  libraries with their with full path.
+
+  Note: On Ubuntu Focal (20.4 LTS) system, the installed CMake export file is broken.
```

Comment:
Probably its older versions as well (the previous LTS also breaks it, so most likely almost all of them don't work).